src/relay/pass/type_solver.cc

```diff
@@ -110,9 +275,10 @@ void TypeSolver::AddConstraint(const TypeConstraint& constraint) {
 
 // Resolve a type in the solver context.
 Type TypeSolver::Resolve(const Type& type) {
+  Resolver resolver(this);
   auto it = tmap_.find(type);
   if (it != tmap_.end()) {
-    return it->second->FindRoot()->resolved_type;
+    return resolver.Resolve(it->second->FindRoot()->resolved_type);
   } else {
     return type;
```

Comment:
If I add a call to the visitor on this line, I get a lot of test failures related to the type inferencer and don't understand why. Is there any reason there would be a problem from visiting a type that's not in the type map?